runelite-client/src/main/java/net/runelite/client/plugins/questtab/QuestTabPlugin.java

```diff
@@ -0,0 +1,614 @@
+
+		if (config.showLength())
+		{
+			// put a colored letter at the beginning of the quest list item text to signify quest length
+			// S - Green - short
+			// M - Yellow - medium
+			// L - Orange - long
+			// V - Red - very long
+			switch (q.getQuestLength())
+			{
+				case SHORT:
+					s.insert(0, "<col=DC10D>S</col> ");
+					break;
+				case MEDIUM:
+					s.insert(0, "<col=FFFF00>M</col> ");
+					break;
+				case LONG:
+					s.insert(0, "<col=FF8C00>L</col> ");
+					break;
+				case VERY_LONG:
+					s.insert(0, "<col=FF0000>V</col> ");
+					break;
+			}
+		}
+
+		if (config.showDifficulty())
+		{
+			// put a colored letter at the beginning of the quest list item text to signify quest difficulty
+			// N - Green - novice
+			// I - Yellow-Green - intermediate
+			// E - Yellow - experienced
+			// M - Orange - master
+			// G - Red - grandmaster
+			switch (q.getQuestDifficulty())
+			{
+				case NOVICE:
+					s.insert(0, "<col=DC10D>N</col> ");
+					break;
+				case INTERMEDIATE:
+					s.insert(0, "<col=9ACD32>I</col> ");
+					break;
+				case EXPERIENCED:
+					s.insert(0, "<col=FFFF00>E</col> ");
+					break;
+				case MASTER:
+					s.insert(0, "<col=FF8C00>M</col> ");
+					break;
+				case GRANDMASTER:
+					s.insert(0, "<col=FF0000>G</col> ");
+					break;
+			}
+		}
+
+		w.setText(s.toString());
+	}
+
+	private void rearrangeWidgets(LinkedHashMap<Quest, Widget> l)
+	{
+		int yPos = QUEST_LIST_TITLE_SPACE;
+
+		for (Widget w : l.values())
+		{
+			w.setRelativeY(yPos);
+			w.setOriginalY(yPos);
+			yPos += QUEST_LIST_ITEM_SPACE;
+		}
+	}
+
+	private static LinkedHashMap<Quest, Widget> sortQuest(HashMap<Quest, Widget> unsortedQuest, SortType s)
+	{
+		List<HashMap.Entry<Quest, Widget>> list = new ArrayList<>(unsortedQuest.entrySet());
+
+		switch (s)
+		{
+			case ALPHABETICAL:
+				// natural order is alphabetical, so sorting by relative y preserves natural/alphabetical order
+				list.sort(Comparator.comparing(a -> a.getValue().getRelativeY()));
+				break;
+			case DIFFICULTY:
+				list.sort(Comparator.comparing(a -> a.getKey().getQuestDifficulty()));
+				break;
+			case LENGTH:
+				list.sort(Comparator.comparing(a -> a.getKey().getQuestLength()));
+				break;
+		}
+
+		LinkedHashMap<Quest, Widget> result = new LinkedHashMap<>();
+
+		for (HashMap.Entry<Quest, Widget> entry : list)
+		{
+			result.put(entry.getKey(), entry.getValue());
+		}
+
+		return result;
+	}
+
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if (!config.menuOption())
+		{
+			return;
+		}
+
+		int groupId = event.getActionParam1();
+		String option = event.getOption();
+
+		if (groupId == WidgetInfo.FIXED_VIEWPORT_QUESTS_TAB.getId() ||
+			groupId == WidgetInfo.RESIZABLE_VIEWPORT_QUESTS_TAB.getId() ||
+			groupId == WidgetInfo.RESIZABLE_VIEWPORT_BOTTOM_LINE_QUEST_TAB.getId())
+		{
+			insertMenuEntry(event, RESET);
+			if (config.sortLength()) insertMenuEntry(event, UNSORT_BY_LENGTH);

```

Comment:
You could probably write a helper function for all these if statements.  Maybe even have all these functions in a list and just iterate through it.